js/webrtc.js

```diff
@@ -881,7 +879,7 @@ var spreedPeerConnectionTable = [];
 
 			var screenNameIndicator = $(screen).find('.nameIndicator');
 
-			if (data.name.length === 0) {
+			if (!data.name || data.name.length === 0) {
```

Comment:
get rid of ` || data.name.length === 0` all strings that match it, also match `!data.name`